source/uwp/SharedRenderer/lib/XamlHelpers.cpp

```diff
@@ -268,21 +283,98 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering::XamlHelpers
+                stream.Seek(0);
+                auto sizeParseOp{ParseSizeOfSVGImageFromStreamAsync(stream)};          
+                co_await wil::resume_foreground(GetDispatcher(strongImageSource));
+                auto size = co_await sizeParseOp;
+                strongImageSource.RasterizePixelHeight(size.Height);
+                strongImageSource.RasterizePixelWidth(size.Width);
+                co_await strongImageSource.SetSourceAsync(stream); 
+                winrt::Image image;
+                image.Source(strongImageSource);
+                if (auto strongTileControl = weakTileControl.get())
+                {
+                    strongTileControl.LoadImageBrush(image);
+                }
+            }
+            co_return;
+        }
+
+        winrt::HttpClient httpClient;
+        auto getOperation = co_await httpClient.GetAsync(uri);
+        auto readOperation = co_await getOperation.Content().ReadAsStringAsync();
+        auto size{ParseSizeOfSVGImageFromXmlString(readOperation)};
+
+        if (auto strongImageSource = weakImageSource.get())
+        {
+            co_await wil::resume_foreground(GetDispatcher(strongImageSource));
+            strongImageSource.RasterizePixelHeight(size.Height);
+            strongImageSource.RasterizePixelWidth(size.Width);
+            winrt::Image image; 
+            image.Source(strongImageSource);
+            strongImageSource.UriSource(imageUrl);
+            if (auto strongTileControl = weakTileControl.get())
+            {
+                strongTileControl.LoadImageBrush(image);
+			}
+        }
+    }
+
     void ApplyBackgroundToRoot(winrt::Panel const& rootPanel,
                                winrt::AdaptiveBackgroundImage const& adaptiveBackgroundImage,
                                winrt::AdaptiveRenderContext const& renderContext)
     {
+        // Creates the background image for all fill modes
+        auto tileControl = winrt::make<winrt::implementation::TileControl>();
+
         // In order to reuse the image creation code paths, we simply create an adaptive card
         // image element and then build that into xaml and apply to the root.
-        if (const auto backgroundImage = CreateBackgroundImage(renderContext, adaptiveBackgroundImage.Url()))
+        if (const auto backgroundImage = CreateBackgroundImage(renderContext, adaptiveBackgroundImage.Url(), tileControl))
         {
-            // Creates the background image for all fill modes
-            auto tileControl = winrt::make<winrt::implementation::TileControl>();
-
             // Set IsEnabled to false to avoid generating a tab stop for the background image tile control
             tileControl.IsEnabled(false);
             tileControl.BackgroundImage(adaptiveBackgroundImage);
-            tileControl.LoadImageBrush(backgroundImage);
+            auto imageUrl = GetUrlFromString(renderContext.HostConfig(), adaptiveBackgroundImage.Url());
```

Comment:
I see a duplicated code, please take LN370 to 372, and make an inline function that returns a boolean. please replace here with a function call and elsewhere. 
please call this function before line 365, save the boolean, pass into CreateBackgroundImage, so we don't have to call the function again.  
also we don't need to turn imageUrl to string, it can be expensive for url to base64 image. please do, std::wstring_view urlWstring {imageUrl.Path};  then call 'find' on urlWstring urlWstring.find(L"svg");